safety/scan/main.py

```diff
@@ -211,13 +253,64 @@ def process_files(paths: Dict[str, Set[Path]], config: Optional[ConfigModel] = N
     if not config:
         config = ConfigModel()
 
-    for file_type_key, f_paths in paths.items():
-        file_type = FileType(file_type_key)
-        if not file_type or not file_type.ecosystem:
-            continue
-        for f_path in f_paths:
-            with InspectableFileContext(f_path, file_type=file_type) as inspectable_file:
-                if inspectable_file and inspectable_file.file_type:
-                    inspectable_file.inspect(config=config)
-                    inspectable_file.remediate()
-                    yield f_path, inspectable_file
+    # old GET implementation
+    if not use_server_matching:
+        for file_type_key, f_paths in paths.items():
+            file_type = FileType(file_type_key)
+            if not file_type or not file_type.ecosystem:
+                continue
+            for f_path in f_paths:
+                with InspectableFileContext(f_path, file_type=file_type) as inspectable_file:
+                    if inspectable_file and inspectable_file.file_type:
+                        inspectable_file.inspect(config=config)
+                        inspectable_file.remediate()
+                        yield f_path, inspectable_file
+
+    # new POST implementation
+    else:
+        files_metadata = []
+        repo_path = os.getcwd()
```

Comment:
I think we should use the target path passed to the `scan` command and pass it as argument to this call.

Reply:
switched it now